### PR TITLE
Add/Expose String.EndsWith(char) 

### PR DESCRIPTION
**Pull request by brfalcon**
* Added overloads for both, StartsWith and EndsWith
* Added all three versions (char, char + bool + culture, char + comparison)
* Renamed the current internal EndsWith(char) to EndsWithOrdinal and kept it internal

### CODE REVIEW

**Comment by danmosemsft** [comment]
Issue is https://github.com/dotnet/corefx/issues/4805

File: src/mscorlib/src/System/String.Comparison.cs [hunk 1]
```diff
@@ -792,6 +792,25 @@ public int CompareTo(String strB)
             return false;
         }
 
+        [Pure]
+        public Boolean EndsWith(Char value)
+        {
+            return EndsWith(new string(value, 1), StringComparison.CurrentCulture);
```
Comment (justinvp) [inline src/mscorlib/src/System/String.Comparison.cs]: Ideally these would not allocate a new string.

**Comment by justinvp** [comment]
The new public APIs need to be added to https://github.com/dotnet/coreclr/blob/master/src/mscorlib/model.xml

File: src/mscorlib/src/System/String.Comparison.cs (hunk 1 above)
Comment (fujiy) [inline src/mscorlib/src/System/String.Comparison.cs]: I couldn't find a way to "culture-aware" compare char's , like CompareInfo do for String. Do we need to create a new API?

Comment (jkotas) [inline src/mscorlib/src/System/String.Comparison.cs]: If the implementation of this API allocates garbage, the API is useless.

Comment (brfalcon) [inline src/mscorlib/src/System/String.Comparison.cs]: @jkotas, Isn't this the same approach of instantiating a new string while calling for the overloaded method in the class CompareInfo, line 756, (https://github.com/dotnet/coreclr/blob/b638af3a4dd52fa7b1ea1958164136c72096c25c/src/mscorlib/src/System/Globalization/CompareInfo.cs#L756)?

Comment (fujiy) [inline src/mscorlib/src/System/String.Comparison.cs]: We could use CompareInfo.IndexOf

Comment (mikedn) [inline src/mscorlib/src/System/String.Comparison.cs]: Do we actually need a culture sensitive `EndsWith` `char` overload? I'd bet that the people who asked for this didn't consider this aspect.

Comment (brfalcon) [inline src/mscorlib/src/System/String.Comparison.cs]: If these overloads aren't needed, we could reuse the implementation of the internal method EndsWithOrdinal(char)..

Comment (justinvp) [inline src/mscorlib/src/System/String.Comparison.cs]: > Do we actually need a culture sensitive EndsWith char overload?

I asked the same thing in the issue: https://github.com/dotnet/corefx/issues/4805#issuecomment-270718228

**Comment by danmosemsft** [comment]
@dotnet-bot test Windows_NT x64 Debug Build and Test  (log gone)

**Comment by danmosemsft** [comment]
@brfalcon do you plan to continue with this PR?

**Comment by brfalcon** [comment]
Hello @danmosemsft ! 
Yes sir :) . Though I'm waiting on a definition about the necessity of reproducing the behaviors requested. This is being discussed at [https://github.com/dotnet/corefx/issues/4805](url)

**Comment by fujiy** [comment]
I created a new PR, https://github.com/dotnet/coreclr/pull/9228

**Comment by jkotas** [comment]
We can close this one then.

**Pull request closed without merging**